backend/pkg/api/connect/service/transform/service.go

```diff
@@ -0,0 +1,164 @@
+		)
+	}
+
+	transforms, err := s.redpandaSvc.ListWasmTransforms(ctx)
+	if err != nil {
+		return nil, apierrors.NewConnectError(
+			connect.CodeInternal,
+			err,
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_REDPANDA_ADMIN_API_ERROR.String()),
+		)
+	}
+	preFilter, err := transformsConversion(transforms)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Msg.GetFilter().GetName() == "" {
+		return &connect.Response[v1alpha1.ListTransformsResponse]{
+			Msg: &v1alpha1.ListTransformsResponse{
+				Transforms: preFilter,
+			},
+		}, nil
+	}
+
+	transform, err := findTransformByName(preFilter, c.Msg.GetFilter().GetName())
+	if err != nil {
+		return nil, err
+	}
+	return &connect.Response[v1alpha1.ListTransformsResponse]{
+		Msg: &v1alpha1.ListTransformsResponse{
+			Transforms: []*v1alpha1.TransformMetadata{transform},
+		},
+	}, nil
+}
+
+// GetTransform gets a transform by name
+func (s *Service) GetTransform(ctx context.Context, c *connect.Request[v1alpha1.GetTransformRequest]) (*connect.Response[v1alpha1.GetTransformResponse], error) {
+	if !s.cfg.Redpanda.AdminAPI.Enabled {
+		return nil, apierrors.NewConnectError(
+			connect.CodeUnimplemented,
+			errors.New("the redpanda admin api must be configured to get a transform"),
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_REDPANDA_ADMIN_API_ERROR.String()),
+			apierrors.NewHelp(apierrors.NewHelpLinkConsoleReferenceConfig()),
+		)
+	}
+
+	transforms, err := s.redpandaSvc.ListWasmTransforms(ctx)
+	if err != nil {
+		return nil, apierrors.NewConnectError(
+			connect.CodeInternal,
+			err,
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_REDPANDA_ADMIN_API_ERROR.String()),
+		)
+	}
+
+	tfs, err := transformsConversion(transforms)
+	if err != nil {
+		return nil, err
+	}
+
+	transform, err := findTransformByName(tfs, c.Msg.Name)
+	if err != nil {
+		return nil, err
+	}
+	return &connect.Response[v1alpha1.GetTransformResponse]{
+		Msg: &v1alpha1.GetTransformResponse{
+			Transform: transform,
+		},
+	}, nil
+}
+
+// DeleteTransform deletes a transform by name
+func (s *Service) DeleteTransform(ctx context.Context, c *connect.Request[v1alpha1.DeleteTransformRequest]) (*connect.Response[v1alpha1.DeleteTransformResponse], error) {
+	if !s.cfg.Redpanda.AdminAPI.Enabled {
+		return nil, apierrors.NewConnectError(
+			connect.CodeUnimplemented,
+			errors.New("the redpanda admin api must be configured to get a transform"),
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_FEATURE_NOT_CONFIGURED.String()),
+			apierrors.NewHelp(apierrors.NewHelpLinkConsoleReferenceConfig()),
+		)
+	}
+
+	transforms, err := s.redpandaSvc.ListWasmTransforms(ctx)
+	if err != nil {
+		return nil, apierrors.NewConnectError(
+			connect.CodeInternal,
+			err,
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_REDPANDA_ADMIN_API_ERROR.String()),
+		)
+	}
+
+	tfs, err := transformsConversion(transforms)
+	if err != nil {
+		return nil, err
+	}
+
+	transform, err := findTransformByName(tfs, c.Msg.Name)
```

Comment:
You're going to want to use a different predicate for list vs get. get and delete should use the same predicate.

Reply:
ok, added new func findExact which uses equality rather than contains